src/libraries/System.Private.DataContractSerialization/src/System/Xml/XmlStreamNodeWriter.cs

```diff
@@ -361,7 +362,7 @@ protected unsafe int UnsafeGetUTF8Chars(char* chars, int charCount, byte[] buffe
                 fixed (byte* _bytes = &buffer[offset])
                 {
                     // Fast path for small strings, use Encoding.GetBytes for larger strings since it is faster when vectorization is possible
-                    if ((uint)charCount < 32)
+                    if (!Vector128.IsHardwareAccelerated || (uint)charCount < 32)
```

Comment:
Where did the "32" come from?  `Vector128<ushort>.Count` is 8 and `Vector256<ushort>.Count` is 16, so this is either 2x a 256-bit vector or 4x a 128-bit vector.  Should it be `if (!Vector128.IsHardwareAccelerated || charCount < Vector128<ushort>.Count`?

Reply:
The number is based on local benchmarks, it was originally 25 but was rounded up 32 after one of the datacontract microbenchmarks had a slight regression for me.

I hope to decrease the limit further after some more performance improvements to the ascii encoding.

It currently happens to be 2× sizeof(Vector128) (same as current limit of for hardware instrincts)